### PR TITLE
feat: Add support for unnest with lateral join

### PR DESCRIPTION
**Pull request by gaurav274**

### CODE REVIEW

File: eva/catalog/schema_utils.py [hunk 1]
```diff
@@ -40,15 +40,15 @@ def get_petastorm_column(df_column):
...
+                np.int64,
                 (),
                 ScalarCodec(IntegerType()),
                 column_is_nullable,
             )
         elif column_type == ColumnType.FLOAT:
             petastorm_column = UnischemaField(
                 column_name,
-                np.float64,
+                np.float32,
```
Comment (xzdandy) [inline eva/catalog/schema_utils.py:51]: What is the motivation of these changes?

File: eva/parser/parser_visitor/_table_sources.py [hunk 2]
```diff
@@ -176,3 +186,12 @@ def visitFromClause(self, ctx: evaql_parser.FromClauseContext):
             where_clause = self.visit(ctx.whereExpr)
 
         return {"from": from_table, "where": where_clause}
+
+    def visitAliasClause(self, ctx: evaql_parser.AliasClauseContext):
+        alias_name = self.visit(ctx.uid())
+        column_list = []
+        if ctx.uidList():
+            column_list = self.visit(ctx.uidList())
```
Comment (xzdandy) [inline eva/parser/parser_visitor/_table_sources.py:194]: I feel it becomes harder over time to figure out which `visit` function it actually calls. Any idea to improve the readability?

File: eva/catalog/schema_utils.py (hunk 1 above)
Comment (gaurav274) [reply eva/catalog/schema_utils.py:51]: I changed it to python's default `int` and `float` types.

File: eva/executor/executor_utils.py [hunk 3]
```diff
@@ -18,6 +18,10 @@
 from eva.models.storage.batch import Batch
 
 
+class ExecutorError(Exception):
```
Comment (xzdandy) [inline eva/executor/executor_utils.py:21]: Where is the `ExecutorError` used?

File: eva/parser/parser_visitor/_table_sources.py (hunk 2 above)
Comment (gaurav274) [reply eva/parser/parser_visitor/_table_sources.py:194]: It is actually quite intuitive, thanks to antlr.
`self.visit(ctx.uidList()) -> visitUidList`

File: eva/executor/executor_utils.py (hunk 3 above)
Comment (gaurav274) [reply eva/executor/executor_utils.py:21]: Not being used right now. I added it in the hope of using it for reporting error messages during execution.

File: test/expression/test_function_expression.py [hunk 4]
```diff
@@ -57,7 +57,7 @@ def test_should_execute_same_function_if_no_gpu(self, context):
 
         context_instance.gpu_device.return_value = NO_GPU
 
-        expression = FunctionExpression(mock_function, name="test")
+        expression = FunctionExpression(mock_function, name="test", alias="func_expr")
```
Comment (xzdandy) [inline test/expression/test_function_expression.py:60]: alias is not `str`?

File: test/integration_tests/test_select_executor.py [hunk 5]
```diff
@@ -246,24 +245,120 @@ def test_select_and_sample(self):
...
+                {
+                    "myvideo.id": np.array([0, 0, 1, 1]),
+                    "T.label": np.array(["person", "person", "bicycle", "bicycle"]),
+                }
+            )
+        )
+
+        self.assertEqual(unnest_batch, expected)
+
+    def test_should_raise_error_with_missing_alias_in_lateral_join(self):
```
Comment (xzdandy) [inline test/integration_tests/test_select_executor.py:312]: So `JOIN LATERAL` much have `AS`. However, it seems from https://github.com/georgia-tech-db/eva/blob/3c16181b140bc3b8cd26e99212d73f14a479ba30/eva/binder/statement_binder.py#L281 , when the node does not have alias, it will use the default node name and column name.

File: eva/parser/parser_visitor/_table_sources.py (hunk 2 above)
Comment (xzdandy) [reply eva/parser/parser_visitor/_table_sources.py:194]: I see it is somehow in the `_insert_statements.py`. I missed it.

File: test/integration_tests/test_select_executor.py (hunk 5 above)
Comment (gaurav274) [reply test/integration_tests/test_select_executor.py:312]: Yes, I have added the code to handle it. But later felt that avoiding it, for now, is better. We can later remove the check from the parser and test it more.

File: test/expression/test_function_expression.py (hunk 4 above)
Comment (gaurav274) [reply test/expression/test_function_expression.py:60]: yeah, it should be `Alias`

**Review by xzdandy — APPROVED** [review]
LGTM

**Comment by gaurav274** [comment]
#144
